### PR TITLE
[TIMOB-6788] Update the JSCore project to compile correctly against iOS 5 SDK

### PR DESCRIPTION
**Pull request by sptramer**
Made some build script changes too, so that they're less cluttered and easier to compile against debug.

Test in JIRA.

### CODE REVIEW

**Comment by WhichKatieDid** [comment]
Code reviewed. ACCEPTED.

File: TiCore/TiCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2560,6 +2560,10 @@
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
 				ARCHS = "$(ARCHS_STANDARD_32_BIT)";
+				"ARCHS[sdk=iphoneos*]" = (
+					armv6,
+					"$(ARCHS_STANDARD_32_BIT)",
+				);
 				ARCHS_iphoneos = "$(ARCHS_STANDARD_32_BIT)";
```
Comment (vishalduggal) [inline TiCore/TiCore.xcodeproj/project.pbxproj]: Is there a reason ARCHS_iphoneos is not updated?

**Comment by vishalduggal** [comment]
Reviewed. **ACCEPTED**

**Comment by vishalduggal** [comment]
Functional Passed. **ACCEPTED**

**Comment by WhichKatieDid** [comment]
Reviewed. ACCEPTED.

**Pull request merged**